### ISSUE TITLE
RFC: add -connection-info flag to build command

### ISSUE THREAD

**Issue by rickard-von-essen** [hashicorp/packer#5107]
We could add a new flag to the build command, say `-connection-info` 
which prints out connection details, passwords, and private keys, etc in the same way as debug does but without the confirmation of each step and disabling of parallelism. `-debug` would implicitly enable this. 

The code changes would be very easy. 

Closes #3885

**Comment by mooperd** [comment]
+1

**Comment by pmkane** [comment]
+1.  This would compliment -on-error=abort/ask nicely, since without the ephemeral ssh key used for remote builds, you're stuck trying to debug without other workarounds.

**Comment by russellpierce** [comment]
This issue does not solve the problems raised in https://github.com/hashicorp/packer/issues/1687.  

To quote that issue:

> --debug behaves exactly as it does now, except it only pauses when an error is encountered

... that means "don't tear down the house after a provisioner failure".  The 'don't pause' and 'show connection info without pausing' aren't the important part in that issue.

**Comment by lmayorga1980** [comment]
:+1:

**Comment by rickard-von-essen** [comment]
Going to start working on this now..

**Comment by justinmetz** [comment]
Hi, are there any updates on this feature?  Sorry if I sound impatient, but I'm really looking forward to it.. :)

**Comment by rickard-von-essen** [comment]
Got a bit derailed, but I'm on it again. Hopefully I can publish a initial version that can be tested on some builders tomorrow.

**Comment by rickard-von-essen** [comment]
@justinmetz I dumped a initial WIP version which works with the `amazon-ebs` builder, if you want to test it out and give me feedback it would be awesome. If you want support for another builder I can hopefully fix that really quickly.

**Comment by suntong** [comment]
I saw that "WIP: Basic support for -connection-info flag #6863" has been closed. So is the `-connection-info` flag feature merged yet? 

Not yet in `Packer v1.5.5`, right?

**Comment by SwampDragons** [comment]
I closed it because it had been an open WIP PR for a year with no further work. It's not merged, but much of the connection info can now be printed using the [build](https://www.packer.io/docs/templates/engine/#build) template engine.

**Comment by testworksau** [comment]
I think there's still a need to be able to show the connection details (eg. password) prior to getting to the point of being able to use the build template engine.

Let's consider the scenario when we need to debug winrm communicator failures in the `amazon-ebs` builder. We need access to the password in order to be able to connect to the machine, but as I understand the password is only output using the `-debug` flag. 

The `-debug` flag cannot be used in an automated build / ci system as it will pause before it attempts to run the `user_data_file`.

I thought `PACKER_LOG=1` would display the password, but it doesn't.

**Comment by kmaehashi** [comment]
Are there any updates on this? Currently, I'm using `packer build -debug -machine-readable` as a workaround to get connection info without pause at each step but would love to have human readable outputs.

**Comment by Glyphack** [comment]
The following page might be helpful to get connection info.
https://developer.hashicorp.com/packer/docs/templates/hcl_templates/contextual-variables

**Comment by suntong** [comment]
Anything specific?

I took a quick look and only found this:

> Build variables will allow you to access connection information and basic instance state information for a builder.

while the OP asked for _"prints out connection details, passwords, and private keys, etc"_

**Comment by Glyphack** [comment]
Yeah I'm not proposing this as a solution but as a work around to echo the `SSHPrivateKey` as part of provisioning.

**Comment by takeda** [comment]
I'm reading these tickets, and I see huge discussions about this, but all people were asking is to have same exact behavior as `-debug`, without unnecessary pauses, that means:

- save the SSH key in current directory
- don't ask to press enter on every little thing packer is doing
- when an error is encountered remind the user the instance's IP and location of the key and wait for pressing enter
- at this time user can ssh and inspect what happened
- after enter is pressed the cleanup begins without asking to confirm every little thing

That's it.

All the `-on-error` and `-connection-info` are unnecessary. This could all be implemented in extending `-debug` parameter to provide extra options:

`-debug=sane` - do what people really want
`-debug=please-waste-my-time` - do the original behavior

**Comment by suntong** [comment]
> all people were asking is to have same exact behavior as `-debug`, without unnecessary pauses

right on!!

**Comment by testworksau** [comment]
> `-debug=sane` - do what people really want
> `-debug=please-waste-my-time` - do the original behavior

😂

**Comment by JrInfra** [comment]
Hello, is there any update on this?  Was thinking this would be nice to have whilst debugging builds without having to constantly press enter.
